### PR TITLE
Create GNIRS Step from Sequence values

### PR DESCRIPTION
**Pull request by jluhrs**
Included several changes made after testing with the real instrument. GNIRS showed several bugs in the code, specially in the ACM assumptions on the order of events.
Also, I had to abandon my intention of triggering all commands from the top level apply record, because I found the corresponding CAR record does not have its CLID connected.

### CODE REVIEW

File: modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/gnirs/Gnirs.scala [hunk 1]
```diff
@@ -132,14 +135,14 @@ object Gnirs {
 
 
   private def autoFilter(w: Length): GnirsController.Filter2 = {
-    val table = Map(
+    val table = List(
```
Comment (jluhrs) [inline modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/gnirs/Gnirs.scala]: What was I thinking ?!

File: modules/edu.gemini.seqexec.server/src/main/resources/Gnirs.xml [hunk 2]
```diff
@@ -85,6 +85,13 @@
                 <description>Filter2</description>
             </parameter>
         </command>
+    </Apply>
+    <Apply name="nirs::dcapply">
```
Comment (cquiroz) [inline modules/edu.gemini.seqexec.server/src/main/resources/Gnirs.xml]: Why's the top `nirs`?

File: modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/gnirs/Gnirs.scala [hunk 3]
```diff
@@ -132,14 +135,14 @@ object Gnirs {
...
       GnirsController.Filter2.J -> 1.42,
       GnirsController.Filter2.H -> 1.86,
-      GnirsController.Filter2.J -> 2.70,
+      GnirsController.Filter2.K -> 2.70,
       GnirsController.Filter2.L -> 4.30,
       GnirsController.Filter2.M -> 6.0
-    ).mapValues(_.nanometers)
+    ).map{ case (f, w) => (f, w.nanometers) }
 
     table.foldRight[GnirsController.Filter2](GnirsController.Filter2.XD){ case (t, v) => if(w < t._2) t._1 else v}
```
Comment (cquiroz) [inline modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/gnirs/Gnirs.scala]: Why do you need to do this?

File: modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/gnirs/Gnirs.scala [hunk 4]
```diff
@@ -175,9 +178,8 @@ object Gnirs {
     case SlitWidth.SW_5        => GnirsController.SlitWidth.Slit0_45.some
     case SlitWidth.SW_6        => GnirsController.SlitWidth.Slit0_68.some
     case SlitWidth.SW_7        => GnirsController.SlitWidth.Slit1_00.some
+    case SlitWidth.PUPIL_VIEWER => GnirsController.SlitWidth.PupilViewer.some
```
Comment (cquiroz) [inline modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/gnirs/Gnirs.scala]: I bet the misalignment bothers you too

File: modules/edu.gemini.seqexec.server/src/main/resources/Gnirs.xml (hunk 2 above)
Comment (jluhrs) [reply modules/edu.gemini.seqexec.server/src/main/resources/Gnirs.xml]: Gemini NIRS

File: modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/gnirs/Gnirs.scala (hunk 3 above)
Comment (jluhrs) [reply modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/gnirs/Gnirs.scala]: I don't need it, I just wanted to have everything as the proper physical quantity, but was too lazy as to add the `.nanometers` to each value.

**Pull request merged**